mlir/lib/Conversion/TosaToRock/TosaToRock.cpp

```diff
@@ -1552,33 +1544,116 @@ struct AttentionRewritePattern : public OpRewritePattern<tosa::MatMulOp> {
     return success();
   }
 
+  // Validates that a constant mask follows the causal mask pattern:
+  // - Each row i should have zeros at positions 0 through i (lower triangular
+  //   part)
+  // - The upper triangular part (positions > i) has to be all 1's
+  bool isValidCausalMask(Operation *op) const {
+    // Get the constant value
+    DenseElementsAttr constAttr;
+    if (auto tosaConst = dyn_cast<tosa::ConstOp>(op)) {
+      constAttr = dyn_cast<DenseElementsAttr>(tosaConst.getValuesAttr());
+    } else if (auto arithConst = dyn_cast<arith::ConstantOp>(op)) {
+      constAttr = dyn_cast<DenseElementsAttr>(arithConst.getValue());
+    }
+
+    if (!constAttr)
+      return false;
+
+    auto shapedType = cast<ShapedType>(constAttr.getType());
+    auto shape = shapedType.getShape();
+
+    // First two dims must be 1 (broadcast dimensions)
+    if (shape.size() != 4 || shape[0] != 1 || shape[1] != 1)
+      return false;
+
+    // Ensure that element type is integer
+    if (!constAttr.getElementType().isIntOrIndex())
+      return false;
+
+    int64_t seqLen = shape[2];
+    int64_t maxSeqLen = shape[3];
+    auto values = constAttr.getValues<APInt>();
+    for (int64_t row = 0; row < seqLen; ++row) {
+      for (int64_t col = 0; col < maxSeqLen; ++col) {
+        auto val = values[row * maxSeqLen + col].getSExtValue();
+
+        // Validate that the lower triangular portion is all zeros
+        if (col <= row && val != 0)
+          return false;
+
+        // Check that the rest is all just 0's and 1's
+        if (col > row && val != 1)
+          return false;
+      }
+    }
+
+    return true;
+  }
+
   FailureOr<Value> getCausal(Value input) const {
-    auto select = getDefiningNonReshapeOpNonCastOp<tosa::SelectOp>(input);
-    if (select) {
+    DenseSet<mlir::OperationName> opsToSkip {
+      mlir::OperationName(tensor::CollapseShapeOp::getOperationName(), getContext()),
+      mlir::OperationName(tensor::ExpandShapeOp::getOperationName(), getContext()),
+      mlir::OperationName(tosa::CastOp::getOperationName(), getContext())
```

Comment:
nit:
Is it possible to just `StringRef`  for op names ? That would be more succinct